src/postgres/tree/PostgresDatabaseTreeItem.ts

```diff
@@ -61,6 +62,14 @@ export class PostgresDatabaseTreeItem extends AzureParentTreeItem<ISubscriptionC
 
         if (username && password) {
             try {
+                if (!await this.parent.isFirewallConfigured()) {
```

Comment:
I'd rather not check for this up front for two reasons:
1. It could have performance implications (slowing down people that already have the firewall configured)
1. It could block people if our logic is incorrect. For example, it seems like `isFirewallConfigured` only searches for a specific firewall rule named "azureDatabasesForVSCode-publicIp", but couldn't I configure a rule myself on the portal with a different name?

Instead, if you return the GenericTreeItem in the catch block (when a firewall-related error occurs), it should solve both of those concerns.